### PR TITLE
Add subworkflows for ensemblvep and snpeff

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: subworkflows/nf-core/annotation_ensemblvep/main.nf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    vep_cache         //    path: path_to_vep_cache (optionnal)
+
+    main:
+    ENSEMBLVEP(vcf, vep_genome, vep_species, vep_cache_version, vep_cache)
+    ANNOTATION_BGZIPTABIX(ENSEMBLVEP.out.vcf)
+
+    emit:
+    vcf_tbi  = ANNOTATION_BGZIPTABIX.out.gz_tbi // channel: [ val(meta), vcf.gz, vcf.gz.tbi ]
+    reports  = ENSEMBLVEP.out.report          //    path: *.html
+    versions = ENSEMBLVEP.out.versions        //    path: versions.yml
```
Comment (mahesh-panchal) [inline subworkflows/nf-core/annotation_ensemblvep/main.nf:23]: Need to `mix` this with the `ANNOTATION_BGZIPTABIX` versions.

File: subworkflows/nf-core/annotation_snpeff/main.nf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    snpeff_cache   //    path: path_to_snpeff_cache (optionnal)
+
+    main:
+    SNPEFF(vcf, snpeff_db, snpeff_cache)
+    ANNOTATION_BGZIPTABIX(SNPEFF.out.vcf)
+
+    emit:
+    vcf_tbi  = ANNOTATION_BGZIPTABIX.out.gz_tbi // channel: [ val(meta), vcf.gz, vcf.gz.tbi ]
+    reports  = SNPEFF.out.report              //    path: *.html
+    versions = SNPEFF.out.versions            //    path: versions.yml
```
Comment (mahesh-panchal) [inline subworkflows/nf-core/annotation_snpeff/main.nf:21]: Again, need to `mix` version channels here. Perhaps also use `first()` to take the first emitted version yaml.

File: tests/subworkflows/nf-core/annotation_ensemblvep/main.nf [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+include { ANNOTATION_ENSEMBLVEP } from '../../../../subworkflows/nf-core/annotation_ensemblvep/main'
+
+workflow annotation_ensemblvep {
+    input = [
+        [ id:'test' ], // meta map
+        file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
+    ]
+
+    ANNOTATION_ENSEMBLVEP ( input, "WBcel235", "caenorhabditis_elegans", "104", [] )
```
Comment (mahesh-panchal) [inline tests/subworkflows/nf-core/annotation_ensemblvep/main.nf:13]: Need a test supplying every parameter too, no? I don't know the suggestions here on test coverage.

File: tests/subworkflows/nf-core/annotation_snpeff/main.nf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+include { ANNOTATION_SNPEFF } from '../../../../subworkflows/nf-core/annotation_snpeff/main'
+
+workflow annotation_snpeff {
+    input = [
+        [ id:'test' ], // meta map
+        file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
+    ]
+
+    ANNOTATION_SNPEFF ( input, "WBcel235.99", [] )
```
Comment (mahesh-panchal) [inline tests/subworkflows/nf-core/annotation_snpeff/main.nf:13]: All parameter test?

Comment (maxulysse) [reply tests/subworkflows/nf-core/annotation_snpeff/main.nf:13]: yes, no need for more, we can add more in args, this is the minimal

File: tests/subworkflows/nf-core/annotation_ensemblvep/main.nf (hunk 3 above)
Comment (maxulysse) [reply tests/subworkflows/nf-core/annotation_ensemblvep/main.nf:13]: Not sure what you meant here actually

Comment (mahesh-panchal) [reply tests/subworkflows/nf-core/annotation_ensemblvep/main.nf:13]: Aren't the test workflows supposed to cover all scenarios? 
So there should also be another test where the `[]` is replaced with a valid parameter.

File: tests/subworkflows/nf-core/annotation_snpeff/main.nf (hunk 4 above)
Comment (maxulysse) [reply tests/subworkflows/nf-core/annotation_snpeff/main.nf:13]: Ooh I see what you mean, then no I don't want to test for that.
This empty parameter is for already downloaded cache supplied by the user.
It's a edge case

File: tests/config/pytest_modules.yml [hunk 5]
```diff
@@ -1451,6 +1451,19 @@ yara/mapper:
...
+#   - *subworkflows_bam_sort_samtools
+
+# subworkflows/annotation_ensemblvep: &subworkflows_annotation_ensemblvep
+#   - subworkflows/nf-core/annotation_ensemblvep/**
+#   - tests/subworkflows/nf-core/annotation_ensemblvep/**
+
+# subworkflows/annotation_snpeff: &subworkflows_annotation_snpeff
+#   - subworkflows/nf-core/annotation_snpeff/**
+#   - tests/subworkflows/nf-core/annotation_snpeff/**
+
```
Comment (maxulysse) [inline tests/config/pytest_modules.yml:1466]: @nf-core/core any idea when we un-comment that?

**Review by mahesh-panchal — APPROVED** [review]
